manifests/system_hook.pp

```diff
@@ -0,0 +1,55 @@
+# @summary A file hook will run on each event so it's up to you to filter events or projects
+# within a file hook code. You can have as many file hooks as you want. Each file hook will
+# be triggered by GitLab asynchronously in case of an event. For a list of events
+# see the system hooks documentation.
+#
+#
+# @example System hook usage
+#   gitlab::system_hook { 'my_system_hook':
+#     type            => 'post-receive',
+#     source          => 'puppet:///modules/my_module/post-receive',
+#   }
+#
+# @param system_hooks_dir The GitLab shell repos path. This defaults to '/opt/gitlab/embedded/service/gitlab-rails/file_hooks' if not present.
+# @param content Specify the system hook contents either as a string or using the template function. If this paramter is specified source parameter must not be present.
+# @param source Specify a file source path to populate the system hook contents. If this paramter is specified content parameter must not be present.
+define gitlab::global_hook (
+  Stdlib::Absolutepath                          $system_hooks_dir,
+  Optional[String[1]]                           $content          = undef,
+  Optional[Pattern[/^puppet:/]]                 $source           = undef,
+) {
+  if $system_hooks_dir {
+    $hook_path = $system_hooks_dir
+  } elsif $gitlab::system_hooks_dir {
+    $hook_path = $gitlab::system_hooks_dir
+  } else {
+    $hook_path = '/opt/gitlab/embedded/service/gitlab-rails/file_hooks'
```

Comment:
can't we set that as default in the init.pp? that would make the logic in this defined resource way shorter?

Reply:
sure!!